### PR TITLE
[documentation] Explain coloring and detector kwargs in sparse backends

### PR DESCRIPTION
**Pull request by amontoison**
Related to #247 
@gdalle 

I renamed the keyword argument for the coloring algorithm, but I believe it's safe to do so.
Since this option was not documented previously, I don't expect any users to rely on the `alg` option.

### CODE REVIEW

File: docs/src/mixed.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Build a hybrid NLPModel
+# Build an hybrid NLPModel
```
Comment (gdalle) [inline docs/src/mixed.md:1]: nope, it was correct before

File: docs/src/sparse.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+x = rand(T, 2)
+J = jac(nlp, x)
+```
+
+```@example ex1
+x = rand(T, 2)
+H = hess(nlp, x)
+```
+
+The available backends for sparse derivatives (`SparseADJacobian`, `SparseADHessian` and `SparseReverseADHessian`) have keyword arguments `detector` and `coloring` to specify the sparsity pattern detector and the coloring algorithm, respectively.
```
Comment (gdalle) [inline docs/src/sparse.md:34]: It's rather weird to call it `SparseADHessian` versus `SparseReverseADHessian`, a better name would be forward-over-forward vs. forward-over-reverse if I understand correctly. But with a switch to DI those distinctions will disappear

**Review by gdalle — COMMENTED** [review]
Looks good to me, but in general it might be useful to add more `(@ref)` and `(@extref)` (from [DocumenterInterLinks.jl](https://github.com/JuliaLang/Pkg.jl/issues/1585))

File: docs/src/sparse.md (hunk 2 above)
Comment (amontoison) [reply docs/src/sparse.md:34]: I didn't know that Tangi implement `SparseReverseADHessian`.
The name is not good, we should switch to DI.jl asap!

File: docs/src/mixed.md (hunk 1 above)
Comment (amontoison) [inline docs/src/mixed.md:1]:
```suggestion
# Build a hybrid NLPModel
```

**Pull request merged**